cpp/include/gpu_operations.h

```diff
@@ -126,8 +127,106 @@ class GpuOperations {
       exit(1);
     }
   }
-  static Matrix<T> Add(const Matrix<T> &a, const T &scalar);
-  static Matrix<T> Add(const Matrix<T> &a, const Matrix<T> &b);
+  static Matrix<T> Add(const Matrix<T> &a, const T &scalar) {
+    int m = a.rows();
+    int n = a.cols();
+    int lda = m;
+    int ldb = m;
+    int ldc = m;
+
+    float alpha = 1.0;
+    float beta = 1.0;
+
+    const T * h_a = &a(0);
+    Matrix<T> b(m, n);
+    b = Matrix<T>::Constant(m, n, scalar);
+    const T * h_b = &b(0);
+    Matrix<T> h_c(m, n);
+
+    T * d_a;  gpuErrchk(cudaMalloc(&d_a, m * n * sizeof(T)));
+    T * d_b;  gpuErrchk(cudaMalloc(&d_b, m * n * sizeof(T)));
+    T * d_c;  gpuErrchk(cudaMalloc(&d_c, m * n * sizeof(T)));
+
+    gpuErrchk(cudaMemcpy(d_a, h_a, m * n * sizeof(T),
+                         cudaMemcpyHostToDevice));
+    gpuErrchk(cudaMemcpy(d_b, h_b, m * n * sizeof(T),
+                         cudaMemcpyHostToDevice));
+
+    cublasStatus_t stat;
+    cublasHandle_t handle;
+    cublasCreate(&handle);
+    stat = cublasSgeam(handle, CUBLAS_OP_N, CUBLAS_OP_N,
```

Comment:
You should use a wrapper function instead, so that both float and double type can be supported. Right now, we put declarations and definitions of all wrapper functions in gpu_util.h and gpu_util.cc. If the wrapper function wrapping the cublasSgeam and cublasDgeam haven't been implemented, you can add them based on similar implementations of others. Be careful of the function name. We now use prefix Gpu, for example, to wrap API function cublasSgeam, we can use a name like GpuMatrixAdd which is more clear than cublasSgeam.